### DISCUSSION TITLE
Nested CODEOWNERS

### DISCUSSION THREAD

**Discussion by chrislloyd** [community/community#21809] [New to GitHub]
<p>I work in a monolithic Git repo - there are thousands of engineers making commits every day. Singular files that every engineer can update and that need frequent updating are a problem. Files like this end up having lots of merge conflicts and end up being very long. <code>CODEOWNERS</code> is currently that kind of file, which makes the whole feature almost useless to us. I’m hesitant to introduce it.</p>
<p>In <a href="https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners" rel="nofollow noopener">the documentation</a> it says that they “A CODEOWNERS file uses a pattern that follows the same rules used in <a href="https://git-scm.com/docs/gitignore#_pattern_format" rel="nofollow noopener">gitignore</a> files”. I think taking the <code>.gitignore</code> inspiration one step further - and allowing <code>CODEOWNERS</code> to be nested, would solve this problem. There are of course difficulties in  doing so (“which CODEOWNERS takes precedence?”), but I would solve these by following the same conventions as git.</p>
<p><code>CODEOWNERS</code> is generally more useful to large teams, and by allowing them to be nested it would be even more useful to the largest of teams. Thanks!</p>

**Comment by MChevy422** [comment, marked as answer]
<p>Hello <span><a class="mention" href="/chrislloyd">@chrislloyd</a></span>, <br><br>Thank you for your feedback! We’re always working to improve GitHub and the GitHub Community Forum, and we consider every suggestion we receive. I’ve logged your feature request in our internal feature request list. Though I can’t guarantee anything or share a timeline for this, I can tell you that it’s been shared with the appropriate teams for consideration.<br><br>Once again, thank you for your input!<br><br>Greatly appreciated<br><br>MChevy422</p>

**Reply by tonglil** [reply]
Future viewers may want to check out https://github.com/marketplace/actions/codeowners-bot

**Reply by gagoar** [reply]
you can also take a look at https://github.com/gagoar/codeowners-generator

**Reply by rolgalan** [reply]
Hi @MChevy422, any news regarding support for nested CODEOWNERS files in the repo? It's two years since it was added into the feature requests list, it would be great to know if this won't ever be implemented or if is planned to be done at some point :). This would be really helpful in big repos to improve readability and maintainability of the code owners file.

**Reply by tgiachetti-cb** [reply]
re: https://github.com/marketplace/actions/codeowners-bot

This is interesting but does not address a very common use case: merge conflicts with codeowners files with different contents on different CI/CD related branches (ie release, develop).

At the moment we are limited to 3 non-conflicting CODEOWNERS files, one in .github, one in root, and one in docs/

Ideally we should be able to have an arbitrary amount of CODEOWNERS files ( ie .github/release/CODEOWNERS, .github/develop/CODEOWNERS).

**Comment by vleandersson** [comment]
+1 on this. For mono-repos this is a critical feature

**Comment by legendof-selda** [comment]
Any update on this feature? it has been 5 years now.
